### ISSUE TITLE
Ubuntu 18.04 + (Bitwig + VST2) or (Ardour + LV2) => crash

### ISSUE THREAD

**Issue by p0nce** [AuburnSounds/Dplug#411]
On Bitwig: First opening of the UI is OK, but second one crash.
Very easy to reproduce.

**Comment by p0nce** [comment]
Repro script:
``` 
#!/bin/sh
dplug-build --compiler ldc2 -c VST
cp builds/Linux-64b-VST/Witty\ Audio\ Destructatorizer.so $LXVST_PATH/Witty\ Audio\ Destructatorizer.so
gdb bitwig-studio
```

**Comment by p0nce** [comment]
- [x] Is it the latest X11 lock changes => NO
- [x] Does it happen with clipit? => NO (!)
- [ ] Does it happen without UI?

**Comment by p0nce** [comment]
Similar crash in Ardour (tells about X11 BadWindow)

**Comment by p0nce** [comment]
When enableing X11 logging, crash happen at window open instead of close. Which again, tend to point to a race.

**Comment by p0nce** [comment]
I'm trying a x11Window rewrite to force me to review in depth. 
Found potential problems:
- eventloop and timerloop threads are joined after the window has been destroyed, which is wrong and time-dependent

**Comment by p0nce** [comment]
Now fixed. 
Also the first draw problem was a race in ` dplug:gui`  and unrelated to X11. X11Window just helped expose it.

**Comment by p0nce** [comment]
BADFIX he race in ` dplug:gui` was badly fixed, and has been hopefully rightly fixed in #422
